### DISCUSSION TITLE
Has Google's Material Design been considered?

### DISCUSSION THREAD

**Discussion by huntc** [emilk/egui#5263] [Q&A] [answered]
Hi there - very new to this codebase, but it looks like a great project!

My first question is, has any consideration been made in terms of a) applying [Material Design](https://m3.material.io/) to this library; and b) if not, does this library lend itself well to applying Material Design?

**Comment by huntc** [comment]
There is the [material-egui](https://github.com/toastxc/material-egui) project (not presently working), but I was more wondering if there has been consideration of supporting Material Design as a core feature.

**Reply by lucasmerlin** [reply]
It's definitely possible but someone would need to create custom widgets with the material look, ripple effect, etc... which would be a lot of work.

**Reply by huntc** [reply]
I wonder then if there’s a staged approach where the existing widgets could be styled, and then more added as time goes on. My feeling is that the library could be greatly enhanced by providing a better out-of-the-box styling that has industry acceptance. Wdyt?

**Reply by lucasmerlin** [reply, marked as answer]
Definitely. Adjusting the spacing probably makes the biggest difference, material has a lot of margin / padding and whitespace while eguis defaults are very compact (which makes sense for most applications where egui is used today).
Adjusting the spacing accordingly to material guidelines should also be relatively straightforward.
